src/components/PopupWithForm.js

```diff
@@ -16,16 +17,23 @@ export default class PopupWithForm extends Popup {
     return inputValues;
   }
 
-  sendInputValues() {
-    return this._getInputValues();
+  _renderSaving(isSaving) {
+    const submitButton = this._popupForm.querySelector(".modal__save");
+    if (isSaving) {
+      submitButton.textContent = "Saving...";
+    } else {
+      submitButton.textContent = this._defaultSubmitButtonText;
+    }
   }
 
   setEventListeners() {
     super.setEventListeners();
     this._popupForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
+      this._renderSaving(true);
       this._handleFormSubmit(this._getInputValues());
       this._popupForm.reset();
```

Comment:
You need to **reset** the form only   in block `then` not to do that immediately after `submit` because when there is an error the popup will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more